src/firmware/RC_firmware/RC_firmware.ino

```diff
@@ -0,0 +1,235 @@
+  //responding to queries from usb
+  sig_read();
+  
+  /*Serial.println("Mode: ");Serial.println(Mode);Serial.println("R1: ");Serial.println(R1);
+  Serial.println("R2: ");
+  Serial.println(R2);
+  Serial.println("R3: ");
+  Serial.println(R3);*/
+  //Serial.println("linear_x_h: ");Serial.println(linear_x_h);Serial.println("linear_x_l: ");Serial.println(linear_x_l);Serial.println("angular_z_h: ");Serial.println(angular_z_h);
+  
+  
+  //serial_read();
+  if (Mode == -1) { //Auto Mode
+    serial_read(); 
+    
+    convert();
+    drive(linear_x, angular_z);
+    //Serial.print("angular_z_l: ");Serial.print(angular_z_l);Serial.print("linear_x: ");Serial.print(linear_x);Serial.print("angular_z: ");Serial.println(angular_z);
+  }
+  else if (Mode == 0) { //RC Mode
+    //Serial.println(angular_z);
+    linear_x = R1; 
+    angular_z = R2;
+    
+    convert();
+    drive(linear_x, angular_z);
+    //Serial.print("angular_z_l: ");Serial.print(angular_z_l);Serial.print("linear_x: ");Serial.print(linear_x);Serial.print("angular_z: ");Serial.println(angular_z);
+    Serial.flushRX();
+  }
+  else { //STOP MODE
+    linear_x = linear_stop; 
+    angular_z = angular_stop;
+    
+    convert();
+    drive(linear_x, angular_z);
+   // Serial.print("angular_z_l: ");Serial.print(angular_z_l);Serial.print("linear_x: ");Serial.print(linear_x);Serial.print("angular_z: ");Serial.println(angular_z);
+    Serial.flushRX();
+  }
+  //Serial.print("linear_x: ");Serial.print(linear_x);Serial.print(" linear_x: ");Serial.print(linear_x);Serial.print(" angular_z: ");Serial.println(angular_z);
+
+}
+
+void set_off() {
+  int linear_x_mid = 1550; //RX standard - radio signal midpoint
+  int angular_z_mid = 1550; //RY standard - radio signal midpoint
+  
+  // joystick_margin is an error margin for joystick control
+  // e.g. if the joystick is moved just a little bit, it is assumed that no movement 
+  // is desired.
+  linear_x_h = linear_x_mid + joystick_margin; 
+  linear_x_l = linear_x_mid - joystick_margin;
+  
+  angular_z_h = angular_z_mid + joystick_margin; 
+  angular_z_l = angular_z_mid - joystick_margin;
+}
+
+
+void sig_read() {
+  R1 = pulseIn(2, HIGH); // 1140 - 1965 RX LEFT-RIGHT -> turn on spot
+  R2 = pulseIn(3, HIGH); // 1965 - 1140 RY UP-DOWN -> Steer left/right y axis (turn while moving)
+  R3 = pulseIn(4, HIGH); // 1970 - 1115 linear_x UP-DOWN -> forward/backward
+  R4 = pulseIn(5, HIGH); //1970 - 1115 mode
+  
+  //Serial.print("R1: "); Serial.print(R1);Serial.print(" R2: "); Serial.print(R2);Serial.print(" R3: "); Serial.println(R3);Serial.println("Mode: ");Serial.println(Mode);
+  if (R1 < linear_x_h && R1 > linear_x_l) 
+    R1 = linear_stop;
+  else 
+    R1 = map (R1, 1140, 1965, 0, 255);
+
+  if (R2 < angular_z_h && R2 > angular_z_l) 
+    R2 = angular_stop;
+  else 
+    R2 = map (R2, 1140, 1965, 0, 255);
+  
+  //Serial.print("R3: ");Serial.println(R3);
+  if (1100 < R3 && R3 < 1400) 
+    Mode = 1; // STOP mode?
```

Comment:
Is it possible to not make these enums? so people don't have to remember 1 = stop, 0 = auto etc..?